odoo/addons/base/models/ir_model.py

```diff
@@ -156,6 +159,24 @@ def _check_model_name(self):
             if not models.check_object_name(model.model):
                 raise ValidationError(_("The model name can only contain lowercase characters, digits, underscores and dots."))
 
+    @api.constrains('order', 'field_id')
+    def _check_order(self):
+        for model in self:
+            try:
+                model._check_qorder(model.order)  # regex check for the whole clause ('is it valid sql?')
+            except UserError as e:
+                raise ValidationError(e)
```

Comment:
`e` is an exception, not a string!
```suggestion
                raise ValidationError(str(e))
```

Reply:
yet
```py
>>> a=Exception('something')
>>> b=NotImplementedError(a)
>>> raise b
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    raise b
NotImplementedError: something
```
works fine :)

I assume Python does some processing on the exception constructor arg to handle this case. But i'll change it anyway, it's more explicit your way